lib/logger.js

```diff
@@ -0,0 +1,85 @@
+const winston = require('winston')
+const { format, transports } = winston
+const { combine, timestamp, label, printf, errors } = format
+const loggerPrefix = 'zwavejs2mqtt.'
+
+// Make sure every config value is defined
+const sanitizedConfig = (module, config) => {
+  config = config || {}
+  return {
+    module: module !== undefined ? module : '-',
+    enabled: config.enabled !== undefined ? config.enabled : true,
+    level: config.level !== undefined ? config.level : 'info',
+    logToFile: config.logToFile !== undefined ? config.logToFile : false,
+    filename:
+      config.filename !== undefined ? config.filename : 'zwavejs2mqtt.log'
+  }
+}
+
+// Custom logging format:
+const customFormat = config =>
+  combine(
+    errors({ stack: true }),
+    timestamp({ format: 'YYYY-MM-DD HH:mm:ss.SSS' }),
+    format(info => {
+      info.level = info.level.toUpperCase()
+      return info
+    })(),
+    label({ label: config.module.toUpperCase() }),
+    printf(
+      info =>
+        `${info.timestamp} ${info.level} ${info.label || '-'}: ${info.message}`
+    )
+  )
+
+// Custom transports:
+const customTransports = config => {
+  const transportsList = [
+    new transports.Console({
+      format: format.colorize({ all: true }),
+      level: config.level
+    })
+  ]
+  if (config.logToFile) {
+    transportsList.push(
+      new transports.File({
+        format: customFormat(config),
+        filename: config.filename,
+        level: config.level
+      })
+    )
+  }
+  return transportsList
+}
+
+// Helper function to setup a logger:
+const setupLogger = (container, module, config) => {
+  config = sanitizedConfig(module, config)
+  const logger = container.add(loggerPrefix + module) // NOTE: Winston automatically reuses an existing module logger
```

Comment:
@AlCalzone We are using a custom prefix for loggers so we shouldn't touch yours

Reply:
Also we are using a custom containers so I even don't think we have any access to your loggers

Reply:
Then 🤷🏻‍♂️

Reply:
FYI: In the last iteration I changed `module()` to directly return `winston.loggers`, since I prefixed the loggers anyway to separate them and the unwanted behavior was still the same using `new winston.Container()`.

@AlCalzone, do you have any pointers for me where I could best hook into the logging setup of `node-zwave-js`? I want to find out the cause of this, since it may bite us somewhere else too ;-).

Reply:
I found the reason of this behavior and it has nothing to do with this PR since it is already behaving that way on master branch.
The responsible lines are in `ZwaveClient.js` which seem to intentionally control the logging behavior of zwave-js using environment variables:
```javascript
  if (process.env.LOGLEVEL === 'undefined') process.env.LOGLEVEL = null
  process.env.LOGLEVEL = process.env.LOGLEVEL || config.logLevel
  if (process.env.LOGTOFILE || config.logToFile) {
    process.env.LOGTOFILE = 'true'
  }
```
@robertsLando I'm not really sure, what your goal is for this with respect to the configuration in `settings.js`?
Currently there's a logging configuration in the `zwave` config, that controls the logging of zwave-js. Should this be kept as is or fully replaced with the central logging configuration you mentioned to be placed into the `gateway` section?

Reply:
> The responsible lines are in ZwaveClient.js which seem to intentionally control the logging behavior of zwave-js using environment variables:

Reply:
@robertsLando any feedback on how to proceed with the zwave-js logging, since I saw that there are TODOs with a PR mentioned (not sure, if you were referring to this PR here)?
Here's the complete code:
```javascript
function ZwaveClient (config, socket) {
  ...
  // TODO: Handle this by using `logOptions` (PR waiting to be merged)
  if (process.env.LOGLEVEL === 'undefined') process.env.LOGLEVEL = null

  // https://github.com/zwave-js/node-zwave-js/blob/master/packages/core/src/log/shared.ts#L13
  // https://github.com/winstonjs/triple-beam/blob/master/config/npm.js#L14-L15
  process.env.LOGLEVEL = process.env.LOGLEVEL || config.logLevel

  if (process.env.LOGTOFILE || config.logToFile) {
    process.env.LOGTOFILE = 'true'
  }
  ...
```

Reply:
Nope, we need to wait zwave-js 6.0.0 for that. Actually that 'workaround' is the only way to configure zwave-js logging